src/GafferScene/InteractiveRender.cpp

```diff
@@ -217,6 +286,437 @@ void InteractiveRender::parentChanged( GraphComponent *child, GraphComponent *ol
 	}
 }
 
+
+
+//////////////////////////////////////////////////////////////////////////
+// BuildTask implementation
+//
+// We use this tbb::task to traverse the input scene and build the
+// hierarchy for the first time. Recursion is terminated for locations
+// at which scene:visible is set to false, so this task also computes
+// and stores the attributes.
+//
+//////////////////////////////////////////////////////////////////////////
+
+class InteractiveRender::SceneGraphBuildTask : public tbb::task
+{
+
+	public :
+
+		SceneGraphBuildTask( const ScenePlug *scene, const Context *context, SceneGraph *sceneGraph, const ScenePlug::ScenePath &scenePath )
+			:	m_scene( scene ),
+				m_context( context ),
+				m_sceneGraph( sceneGraph ),
+				m_scenePath( scenePath )
+		{
+		}
+
+		~SceneGraphBuildTask()
+		{
+		}
+
+		virtual task *execute()
+		{
+			ContextPtr context = new Context( *m_context, Context::Borrowed );
+			context->set( ScenePlug::scenePathContextName, m_scenePath );
+			Context::Scope scopedContext( context.get() );
+
+			m_sceneGraph->m_attributes = m_scene->attributesPlug()->getValue();
```

Comment:
I don't quite follow why we don't store the hash of the attributes here as well - it seems that we then have to recompute the hash in `SceneGraphEvaluatorFilter( onlyChanged = false )`, and we're also recomputing the attributes there as well. Could be not just reuse what was computed here (if we stored the hash)?